addon/components/column-chart/component.js

```diff
@@ -425,26 +430,57 @@ export default BaseChartComponent.extend({
             }
         });
 
-        for (let i = 0; i < bars.length; i++) {
-            if (!values[i]) {
-                continue;
-            }
+        if (this.get('labelOptions.labelCollisionResolution') === 'auto') {
+            /* auto adjust when to hide data vlue of a bar
+            * if labels overlap
+            */
+            const barWidth = Number(d3.select(bars[0]).attr('width'));
+            // bar gap is differenct abscissa of one bar and  abscissa of next bar, minus the bar width.
```

Comment:
Today I learned that an abscissa is a thing. That said, I don't think this comment is necessary because the variable name + logic explains what you're saying by itself